### PR TITLE
Error handling

### LINKED ISSUE #29: Add Error Handling Component
We need a component to display when there is an error with a Promise's response or the user routes to a wrong URL.

### LINKED ISSUE #30: Implement Error Boundaries
https://reactjs.org/docs/error-boundaries.html

### PR DESCRIPTION
**Pull request by baileydunning**
### What does this pull request do?

+ Installed [react-error-boundary](https://www.npmjs.com/package/react-error-boundary#errorboundary-props) npm package. So when this is merged you'll have to run another `npm i` to update.
  + `App` is now wrapped in an `ErrorBoundary` imported from that package. There are a lot of different props we can utilize, but right now I just implemented the `FallbackComponent` prop (`<Error500 />`). 
  + This is a good [article](https://kentcdodds.com/blog/use-react-error-boundary-to-handle-errors-in-react) that walks through the example in the docs too
+ Created error components for 400/500 status codes. Right now they are super basic components but the idea behind having two error components is to display a more specific message. I read this [Medium article](https://medium.com/thinking-design/how-to-write-design-user-friendly-error-messages-87d0207bb902) about writing / designing user-friendly error messages and thought it was interesting. I'm sure there are some fun ways to tailor it to kids as well.
+ Note that I added a temporary route for `Error500` to be able to see it and work with it

### What files were changed?

+ `App`
+ `modules/common/error`
  + `Error400`
  + `Error500`

### Issues
+ #30 
+ closes #29 

### Next steps
+ Add styling, imagery, figure out what messages we want 
+ `useErrorHandler` - once we get our fetches set up
```
  const handleError = useErrorHandler()
  function handleSubmit(event) {
    event.preventDefault()
    const name = event.target.elements.name.value
    setState({status: 'pending'})
    fetchGreeting(name).then(
      newGreeting => setState({greeting: newGreeting, status: 'resolved'}),
      error => handleError(error),
    )
  }
```
+ `onReset`

```
const ui = (
  <ErrorBoundary
    FallbackComponent={ErrorFallback}
    onReset={() => {
      // reset the state of your app so the error doesn't happen again
    }}
  >
    <ComponentThatMayError />
  </ErrorBoundary>
)
```

### Screenshots / Gifs
400-error
500-error

### CODE REVIEW

File: src/modules/App/App.js [hunk 1]
```diff
@@ -7,20 +7,27 @@ import StyleSample from "../views/StyleSample";
 import DailyMission from "../views/DailyMission";
 import Sidebar from "../common/sidebar/Sidebar";
 import Header from "../common/header/Header";
+import Error400 from "../common/error/Error400";
```
Comment (lo-la-do-li) [inline src/modules/App/App.js:10]: Good call on specifying the different error messages!

File: src/modules/App/App.js [hunk 2]
```diff
@@ -7,20 +7,27 @@ import StyleSample from "../views/StyleSample";
...
+import Error500 from "../common/error/Error500";
+import { ErrorBoundary } from "react-error-boundary";
```
Comment (brabbuss) [inline src/modules/App/App.js:12]: Always been curious about error boundaries in React, thank you for implementing

**Review by brabbuss — APPROVED** [review]
I think this is a great PR. The error handling isn't flashy, but great best practice. Thank you for linking some research in the Medium article, very helpful and I'm on board. I think Error screens are where fun Easter eggs will live on my portfolio site. 

Here's my absolute favorite so far: http://joshwcomeau.com/notapage

File: src/modules/common/error/Error400.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import { useHistory } from "react-router-dom";
+import theme from "../../../ui/common/theme";
+
+const Error400 = () => {
+  const history = useHistory()
+  return (
+    <section style={{color: `${theme.colors.white}`, textAlign: "center"}}>
+      <h1>Looks like you got lost!</h1>
+      <h3>Unfortunately, this page doesn't exist (yet)</h3>
+      <Button onClick={() => history.push("/mission-control")}>Take me back to mission control!</Button>
```
Comment (lo-la-do-li) [inline src/modules/common/error/Error400.jsx:12]: I like the thematic/customized messages :)

**Review by lo-la-do-li — APPROVED** [review]

**Pull request merged**